tests/components/caldav/test_calendar.py

```diff
@@ -173,8 +189,21 @@ def _mock_calendar(name):
     for idx, event in enumerate(EVENTS):
         events.append(Event(None, "%d.ics" % idx, event, None, str(idx)))
 
+    def date_search(start, end):
```

Comment:
Where is this logic defined in the code under test? It would be good to not have to redefine this logic in the test, but use the logic in the code under test, so we make sure we're using the logic in the test that will actually be used in the code under test.

Reply:
This is the method of an object in the `caldav` library, so I don't see how it would be possible to use that directly. This is why I had to "simulate" its effect. In the original test, it was defined to just return all events and I added the date filtering to it.

Reply:
Ok. Since we now use `to_datetime` and `get_end_date` of `WebDavCalendarData` in the setup phase, creating the mocks, of the tests here, we should test these methods separately also.

Please add a couple of tests for those methods specifically. We don't need to set up the component for that. We can test those methods in isolation.

Reply:
Any help on this by anyone following this thread would be appreciated. Not sure when I will have time for that.

Reply:
@MartinHjelmare those methods weren't introduced as part of this PR. testing is important but asking contributors to add tests for code they didn't create is a huge hurdle to contribution as demonstrated by this PR laying stagnant.
here is the blame for that file: https://github.com/home-assistant/home-assistant/blame/5c2c4cb4e6d8ee73cf94c6fb3eb5aa4eae8f6a8e/homeassistant/components/caldav/calendar.py

is there a way we can just focus specifically on what's changed in this PR rather than asking the author who has already spent a lot of time on this to add in more tests for code that wasn't introduced by him?

Reply:
We need to test those methods since we're now using them in the mocking phase of the tests. This was introduced with this PR so we need to handle this in this PR.

Reply:
I agree with @cmsimike and to be honest I find your requests exaggerated. I changed a single digit in the code, which was not tested previously. Admittedly it is not elegant to use these methods in the test, but the main problem was, as you noticed above, simulating the effect of a method from an external library. The way I always understood the developer docs (and the PR guidelines) is that one does not have to add unit tests for code from external libraries. So in my opinion the cleanest solution would be to remove this test (even if it took me some time to write).

Reply:
Since we have existing tests for this integration the procedure is that new tests are added for new features for that integration. I'm not discussing this further.

You're welcome to seek the help of another member to get this PR merged, if they have another opinion.

Reply:
Where is that procedure written @MartinHjelmare? Or are you saying it's procedure because that's what you want? Open source isn't about you, it's about the community.
Shutting down lines of communication shows that you don't have interest in doing what's best for the community, just you. Which sucks for everyone except for you. Now this PR will be delayed further if merged at all because you are throwing a tantrum.